### PR TITLE
New example: bacteria in daycare centers

### PR DESCRIPTION
**Pull request by vuolleko**
Example of inference of transmission dynamics of bacteria in day care centers.

Treatment roughly follows:
Numminen, E., Cheng, L., Gyllenberg, M. and Corander, J.: Estimating the transmission dynamics
of Streptococcus pneumoniae from strain prevalence data, Biometrics, 69, 748-757, 2013.

### CODE REVIEW

File: elfi/examples/daycare.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    t1 = np.asanyarray(t1).reshape((-1, 1, 1, 1))
+    t2 = np.asanyarray(t2).reshape((-1, 1, 1, 1))
+    t3 = np.asanyarray(t3).reshape((-1, 1, 1, 1))
+
+    if freq_strains_commun is None:
+        freq_strains_commun = np.full(n_strains, 0.1)
+
+    prob_commun = t2 * freq_strains_commun
+
+    state = np.zeros((batch_size, n_dcc, n_ind, n_strains), dtype=np.bool)  # infection status
```
Comment (hpesonen) [inline elfi/examples/daycare.py]: Could it be explicitly stated what the dimensions of the state are as the tensor-implementation may be difficult to follow.

File: elfi/examples/daycare.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+        # init prob to get infected, same for all
+        hazards = t1 * prob_strain + prob_commun  # shape (batch_size, n_dcc, 1, n_strains)
+
+        # co-infection, depends on the individual's state
+        hazards = np.tile(hazards, (1, 1, n_ind, 1))
+        any_infection = np.any(state, axis=3, keepdims=True)
+        hazards = np.where(any_infection, t3 * hazards, hazards)
+
+        # prob to be cured
```
Comment (hpesonen) [inline elfi/examples/daycare.py]: These are not probabilities, yet, but maybe, 'simulated frequencies', or sth?

File: elfi/examples/daycare.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+        hazards[state] = gamma
+
+        # normalize to probabilities
+        inv_sum_hazards = 1. / np.sum(hazards, axis=(2, 3), keepdims=True)
+        probs = hazards * inv_sum_hazards
+
+        # time until next transition
```
Comment (hpesonen) [inline elfi/examples/daycare.py]: Maybe use plural 'times' here (in the comment), as these are random times for transitions on the whole system.

File: elfi/examples/daycare.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+
+    prob_commun = t2 * freq_strains_commun
+
+    state = np.zeros((batch_size, n_dcc, n_ind, n_strains), dtype=np.bool)  # infection status
+    time = np.zeros((batch_size, n_dcc))
+    n_factor = 1. / (n_ind - 1)
+    gamma = 1.
+    ind_b_dcc = [np.repeat(np.arange(batch_size), n_dcc), np.tile(np.arange(n_dcc), batch_size)]
+
+    i = 0
```
Comment (hpesonen) [inline elfi/examples/daycare.py]: What's counter i for?

File: elfi/examples/daycare.py (hunk 1 above)
Comment (vuolleko) [reply elfi/examples/daycare.py]: Added some explanation (including missing `n_dcc` in docstring).

File: elfi/examples/daycare.py (hunk 3 above)
Comment (vuolleko) [reply elfi/examples/daycare.py]: Done

File: elfi/examples/daycare.py (hunk 2 above)
Comment (vuolleko) [reply elfi/examples/daycare.py]: Done

File: elfi/examples/daycare.py (hunk 4 above)
Comment (vuolleko) [reply elfi/examples/daycare.py]: Nothing anymore. Removed. :)

**Pull request merged**